packages/functional-tests/src/tests/rigid-body-test.ts

```diff
@@ -11,12 +11,17 @@ import Test from './test';
 
 export default class RigidBodyTest extends Test {
 
+    private material: MRESDK.Material;
+
     constructor(app: App) {
         super(app);
     }
 
     public async run(): Promise<boolean> {
         let success = true;
+        this.material = await this.app.context.assetManager.createMaterial('blue', {
+            color: MRESDK.Color3.Blue().toJSON()

```

Comment:
>.toJSON() [](start = 39, length = 9)

.toJSON() is no longer necessary here.

Reply:
Whoops, didn't mean to check this in! Should I leave the colors on there, or strip them out?

Reply:
Oh wait, it probably it here. It's not necessary in the actor create messages because we resolve them manually.

---
In reply to: [255245732](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/186#discussion_r255245732) [](ancestors = 255245732)

Reply:
It isn't a lot of added complexity. It could stay I think, if you preferred it.

---
In reply to: [255248112](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/186#discussion_r255248112) [](ancestors = 255248112)

Reply:
It's actually important to resolve these values using `resolveJsonValues`. Check out how it's done in `createActorFromPayload`.

---
In reply to: [255248308](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/186#discussion_r255248308) [](ancestors = 255248308,255245732)